pytorch_lightning/strategies/collaborative.py

```diff
@@ -0,0 +1,513 @@
+        # todo: cyclic import, someone show me the way
+        from pytorch_lightning.accelerators.cpu import CPUAccelerator
+        from pytorch_lightning.accelerators.gpu import GPUAccelerator
+
+        if isinstance(self.accelerator, GPUAccelerator):
+            return torch.device(f"cuda:{torch.cuda.current_device()}")
+        elif isinstance(self.accelerator, CPUAccelerator):
+            return torch.device("cpu")
+        raise ValueError
+
+    @property
+    def is_global_zero(self) -> bool:
+        return True
+
+    def setup(self, trainer: "pl.Trainer") -> None:
+        self.model_to_device()
+        super().setup(trainer)
+        if self.precision_plugin.precision in (PrecisionType.HALF, PrecisionType.MIXED):
+            self.precision_plugin.scaler = hivemind.GradScaler()
+
+    def _initialize_hivemind(self) -> None:
+        if len(self.optimizers) > 1:
+            raise MisconfigurationException("Hivemind only supports training with one optimizer.")
+        optimizer = self.optimizers[0]
+
+        if self._require_scheduler_fn and self._scheduler_fn is None:
+            rank_zero_warn(
+                "Enabling delay_optimizer_step, delay_state_averaging or offload_optimizer "
+                "requires a scheduler_fn to be passed to the strategy if a scheduler is being used "
+                "(this is because the optimizer is re-created within Hivemind)."
+            )
+
+        scheduler = self._scheduler_fn if self._require_scheduler_fn else None
+        params = optimizer.param_groups if self._require_scheduler_fn else None
+        optimizer = type(optimizer) if self._require_scheduler_fn else optimizer
+        opt = hivemind.Optimizer(
+            dht=self.dht,
+            run_id=self._run_id,
+            params=params,
+            optimizer=optimizer,
+            scheduler=scheduler,
+            target_batch_size=self._target_batch_size,
+            batch_size_per_step=self._batch_size,
+            **self._optimizer_kwargs,
+        )
+
+        if not self._scheduler_fn:
+            self._wrap_schedulers(opt)
+        opt.load_state_from_peers()
+        self.optimizers = [opt]
+        self._opt = opt
+
+        if self._reuse_grad_buffers:
+            assert self.lightning_module is not None
+            # turn off zero_grad by Lightning
+            if is_overridden("optimizer_zero_grad", self.lightning_module):
+                rank_zero_warn(
+                    "You have overridden `optimizer_zero_grad` which will be disabled. "
+                    "When CollaborativeStrategy(reuse_grad_buffers=True), "
+                    "the optimizer cannot call zero grad, as this would "
+                    "delete the gradients before they are averaged."
+                )
+
+            def override_fn(*args: Any, **kwargs: Any) -> None:
+                pass
+
+            self.lightning_module.optimizer_zero_grad = override_fn  # type: ignore[assignment]
```

Comment:
This *should* work too, and has the benefit that when this is called again later, `is_overridden` should return False

```suggestion
            self.lightning_module.optimizer_zero_grad = None
```